pydicom/pixel_data_handlers/util.py

```diff
@@ -1,39 +1,415 @@
+
+    Returns
+    -------
+    numpy.dtype
+        The numpy data type to be used for the pixel data, considering
+        the endianess.
     """
     if is_little_endian is None:
         raise ValueError("Dataset attribute 'is_little_endian' "
                          "has to be set before writing the dataset")
 
-    if is_little_endian != sys_is_little_endian:
+    if is_little_endian != (byteorder == 'little'):
         return numpy_dtype.newbyteorder('S')
 
     return numpy_dtype
+
+
+def pixel_dtype(ds):
+    """Return a numpy dtype for the pixel data in dataset in `ds`.
+
+    Suitable for use with IODs containing the Image Pixel module.
+
+    +------------------------------------------+--------------+
+    | Element                                  | Supported    |
+    +-------------+---------------------+------+ values       |
+    | Tag         | Keyword             | Type |              |
+    +=============+=====================+======+==============+
+    | (0028,0101) | BitsAllocated       | 1    | 1, 8, 16, 32 |
+    +-------------+---------------------+------+--------------+
+    | (0028,0103) | PixelRepresentation | 1    | 0, 1         |
+    +-------------+---------------------+------+--------------+
+
+    Parameters
+    ----------
+    ds : dataset.Dataset
+        The DICOM dataset containing the pixel data you wish to get the
+        numpy dtype for.
+
+    Returns
+    -------
+    numpy.dtype
+        A numpy dtype suitable for containing the dataset's pixel data.
+
+    Raises
+    ------
+    NotImplementedError
+        If the pixel data is of a type that isn't supported by either numpy
+        or pydicom.
+    """
+    if not HAVE_NP:
+        raise ImportError("Numpy is required to determine the dtype.")
+
+    if ds.is_little_endian is None:
+        ds.is_little_endian = ds.file_meta.TransferSyntaxUID.is_little_endian
+
+    # (0028,0103) Pixel Representation, US, 1
+    #   Data representation of the pixel samples
+    #   0x0000 - unsigned int
+    #   0x0001 - 2's complement (signed int)
+    pixel_repr = ds.PixelRepresentation
+    if pixel_repr == 0:
+        dtype_str = 'uint'
+    elif pixel_repr == 1:
+        dtype_str = 'int'
+    else:
+        raise NotImplementedError(
+            "Unable to determine the data type to use to contain the "
+            "Pixel Data as a value of '{}' for '(0028,0103) Pixel "
+            "Representation' is not supported".format(pixel_repr)
+        )
+
+    # (0028,0100) Bits Allocated, US, 1
+    #   The number of bits allocated for each pixel sample
+    #   PS3.5 8.1.1: Bits Allocated shall either be 1 or a multiple of 8
+    #   For bit packed data we use uint8
+    bits_allocated = ds.BitsAllocated
+    if bits_allocated == 1:
+        dtype_str = 'uint8'
+    elif bits_allocated > 0 and bits_allocated % 8 == 0:
+        dtype_str += str(bits_allocated)
+    else:
+        raise NotImplementedError(
```

Comment:
Same here - this would be illegal DICOM, so I would use something like `ValueError`.